starboard/shared/win32/posix_emu/socket.cc

```diff
@@ -98,11 +113,35 @@ int close(int fd) {
   SOCKET socket_handle = handle_db_get(fd, true);
 
   if (socket_handle != INVALID_SOCKET) {
+    WSACleanup();
     return closesocket(socket_handle);
   }
 
   // This is then a file handle, so use Windows `_close` API.
+  WSACleanup();
```

Comment:
You’re correct. It’s not appropriate to call `WSAStartUp()` and `WSACleanup()` for every socket operation. However, setsockopt() will fail if we don’t invoke WSAStartUp() beforehand. Therefore, I’ve made changes to include <winsock2.h> in the NPLB POSIX compliance tests and to invoke WSAStartUp() and WSACleanup() respectively in SetUpTestSuite() and TearDownTestSuite() for WIN32. @maxz-lab, have you noticed the same issue with other socket tests?

Reply:
Closing this convo as after removing the WSA calls win32 tests still succeeded.

I added the WSA calls before in order to fix the 10038 (WSAENOTSOCK) error,, I must got confused and that error was fixed by something else rather than the WSA calls I added...